### DISCUSSION TITLE
Using extracted fonts in EPUBs

### DISCUSSION THREAD

**Discussion by jwes44** [pymupdf/PyMuPDF#2820] [Q&A]
I am looking into converting PDFs to EBUBs and in some cases I will need to use the extracted fonts. What sort of difficulties will I run into and what tools will help me automate the process as much as possible?

**Comment by JorjMcKie** [comment]
@jwes44 PyMuPDF does not yet support conversion to EPUB.
And I believe MuPDF doesn't either, but you should ask this question in MuPDF's public [Discord](https://discord.gg/stKJn3xt) channel.

**Reply by jwes44** [reply]
PyMuPDF supplies enough information so I can write an epub, but I am not sure if I can use the fonts that are extracted as they are, and I was hoping someone else had tried this.

**Reply by JorjMcKie** [reply]
Ah ok, sorry I did not understand you then.

Advanced user as are one, you are probably aware of the font extraction method doc.extract_font(xref)  and the font buffer it returns as the last item of the returned tuple.

This is the binary image of the font as found in the PDF. Other items of the tuple contain more info that may help to use that buffer, like the recommended file extension, e.g. `ttf`.

Whether you can actually use this extracted data cannot be predicted. I saw it working and I also saw it failing.
As a matter of course, only embedded fonts are eligible anyway.
You also have to expect that these fonts are subset font more often than not. And depending and their making, they should be sufficient to re-render all text contained in the PDF.

If you run into problems, you may be successful in identifying the original full font from the name of font subset, which you then may use instead.

I am afraid that is about the only advice I have for you.
